src/scipp/core/variable.py

```diff
@@ -21,6 +21,7 @@
 
 
 def scalar(value: Any,
+           *,
            variance: Any = None,
            unit: Union[Unit, str, None] = default_unit,
```

Comment:
Yes, could be nice. I don't think this could lead to problems when migrating because `sc.scalar(1.2, 3.3)` which worked until now will raise a `TypeError` with your suggestion and `sc.scalar('a', 'm')` was never supported.